### PR TITLE
[demo] - enable complex service definitions

### LINKED ISSUE #609: Helm scheme doesn't support service annotations for frontendProxy
When configuring the frontendProxy component, It is beneficial to configure the frontendProxy component to support additional annotations for managed services like AWS Load Balancer Controller, such as Network Load Balancer (NLB) with instance or IP targets through a Kubernetes service of type LoadBalancer with relevant annotations. However the scheme does not support this today, and static IPs can not be assigned. 

i.e.
```
service.beta.kubernetes.io/aws-load-balancer-type
service.beta.kubernetes.io/aws-load-balancer-eip-allocations
service.beta.kubernetes.io/aws-load-balancer-subnet
```

### PR DESCRIPTION
**Pull request by puckpuck**
Fixes #609 

Moves `serviceType` and `servicePort` parameters to be members of a `service` parameter as children called `type` and `port`. It also adds support for `annotations` and `nodePort` under the same parent parameter.

This will be a breaking change for people who overrode either of the component service parameters before.

### CODE REVIEW

File: charts/opentelemetry-demo/examples/default/rendered/component.yaml [hunk 1]
```diff
@@ -1812,10 +1764,6 @@ spec:
         - name: shippingservice
           image: 'ghcr.io/open-telemetry/demo:1.3.0-shippingservice'
           imagePullPolicy: IfNotPresent
-          ports:
-          
-          - containerPort: 8080
```
Comment (TylerHelmuth) [inline charts/opentelemetry-demo/examples/default/rendered/component.yaml:1817]: Is the removal of these ports expected?

File: charts/opentelemetry-demo/README.md [hunk 2]
```diff
@@ -27,6 +27,13 @@ helm install my-otel-demo open-telemetry/opentelemetry-demo
 
 ## Upgrading Chart
 
+### To 0.18
```
Comment (TylerHelmuth) [inline charts/opentelemetry-demo/README.md:30]: How do you feel about moving this to an `UPGRADING.md` file?  This is what the other charts use.

File: charts/opentelemetry-demo/examples/default/rendered/component.yaml (hunk 1 above)
Comment (puckpuck) [reply charts/opentelemetry-demo/examples/default/rendered/component.yaml:1817]: no... I need to look into why this happened

File: charts/opentelemetry-demo/README.md (hunk 2 above)
Comment (puckpuck) [reply charts/opentelemetry-demo/README.md]: done

File: charts/opentelemetry-demo/examples/default/rendered/component.yaml (hunk 1 above)
Comment (puckpuck) [reply charts/opentelemetry-demo/examples/default/rendered/component.yaml]: this is fixed now

File: charts/opentelemetry-demo/examples/public-hosted-ingress/rendered/component.yaml [hunk 3]
```diff
@@ -1857,29 +1857,3 @@ spec:
           resources:
             limits:
               memory: 20Mi
----
-# Source: opentelemetry-demo/templates/component.yaml
-apiVersion: "networking.k8s.io/v1"
```
Comment (TylerHelmuth) [inline charts/opentelemetry-demo/examples/public-hosted-ingress/rendered/component.yaml:1862]: @puckpuck I believe the new service object refactor broke the `otel-demo.ingress` template.

**Comment by puckpuck** [comment]
I did a search on the entire code base for `serviceType` and `servicePort` to make sure we aren't referencing the old values anymore and that they are adopted for the new structure.

File: charts/opentelemetry-demo/examples/public-hosted-ingress/rendered/component.yaml (hunk 3 above)
Comment (puckpuck) [reply charts/opentelemetry-demo/examples/public-hosted-ingress/rendered/component.yaml]: fixed

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**